lib/rest-adapter.js

```diff
@@ -133,6 +133,27 @@ RestAdapter.prototype.connect = function(url) {
   this.connection = url;
 };
 
+/**
+ * Get the auth from the remotes or from the
+ * loopback options if available
+ * @param remotes
+ * @param args
+ * @private
+ */
+RestAdapter.prototype._extractAuth = function(remotes, invokeOptions) {
+  var auth = remotes.auth;
+  // check for the options to pass along to the rest endpoint.
+  // It may have the access token that can be used
+  if (auth || !invokeOptions) return auth;
+  if (this.options && this.options.passAccessToken && invokeOptions &&
```

Comment:
Instead of checking if `this.options` is set everywhere, it's usually easier to ensure it's at least an empty object in the constructor.

I have also rearranged the code in this function to make it easier to follow for me (hopefully it still looks clear to you too!)